crates/stackable-versioned-macros/src/codegen/container/struct.rs

```diff
@@ -391,4 +465,52 @@ impl Struct {
             _ => None,
         }
     }
+
+    pub fn generate_kubernetes_status_struct(&self) -> Option<TokenStream> {
+        match &self.common.options.kubernetes_options {
+            Some(kubernetes_options) => {
+                let status_ident = format_ident!(
+                    "{struct_ident}Status",
+                    struct_ident = self.common.idents.kubernetes.as_ident()
+                );
+
+                let versioned_crate = &*kubernetes_options.crates.versioned;
+                let schemars_crate = &*kubernetes_options.crates.schemars;
+                let serde_crate = &*kubernetes_options.crates.serde;
+
+                match &kubernetes_options.status {
+                    Some(status) => Some(quote! {
+                        #[derive(
+                            ::core::clone::Clone,
+                            ::core::fmt::Debug,
+                            #serde_crate::Deserialize,
+                            #serde_crate::Serialize,
+                            #schemars_crate::JsonSchema
+                        )]
+                        #[serde(rename_all = "camelCase")]
+                        pub struct #status_ident {
+                            pub crd_values: #versioned_crate::CrdValues,
+
+                            #[serde(flatten)]
+                            pub status: #status,
+                        }
+                    }),
+                    None => Some(quote! {
```

Comment:
We should combine the variants and only conditionally have
```
                            #[serde(flatten)]
                            pub status: #status,
```

Reply:
Done in c448ed2.